### PR TITLE
Add runtime version MazeRunner tests

### PR DESCRIPTION
**Pull request by tomlongridge**
## Goal

Add tests to cover the runtime version information now stored in device data.

### Added

* features/runtime_versions.feature - new test to check for the presence of a version string

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: features/runtime_versions.feature [hunk 1]
```diff
@@ -0,0 +1,45 @@
+Feature: Reporting runtime versions
+
+Background:
+  Given I set environment variable "BUGSNAG_API_KEY" to "a35a2a72bd230ac0aa0f52715bbdc6aa"
+  And I configure the bugsnag endpoint
+
+Scenario: report for unhandled event contains runtime version information
```
Comment (Cawllec) [inline features/runtime_versions.feature]: This one's the handled event

File: features/runtime_versions.feature [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  And I wait for the app to respond on the appropriate port
+  When I navigate to the route "/handled_exception"
+  And I wait for 1 second
+  Then I should receive a request
+  And the request is a valid for the error reporting API
+  And the event "unhandled" is false
+  And the event "device.runtimeVersions.php" matches "(\d+\.){2}\d+"
+  And the event "device.runtimeVersions.laravel" matches "(\d+\.){2}\d+"
+
+Scenario: report for handled event contains runtime version information
```
Comment (Cawllec) [inline features/runtime_versions.feature]: This one's the unhandled one :)

File: features/runtime_versions.feature [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  And I wait for the app to respond on the appropriate port
+  When I navigate to the route "/unhandled_exception"
+  And I wait for 1 second
+  Then I should receive a request
+  And the request is a valid for the error reporting API
+  And the event "unhandled" is true
+  And the event "device.runtimeVersions.php" matches "(\d+\.){2}\d+"
+  And the event "device.runtimeVersions.laravel" matches "(\d+\.){2}\d+"
+
+Scenario: Sessions are correct in unhandled exceptions from controllers
```
Comment (Cawllec) [inline features/runtime_versions.feature]: "Runtime versions are correct in sessions" ?
